setuptools/tests/test_packageindex.py

```diff
@@ -224,6 +225,41 @@ def test_egg_fragment(self):
             assert dists[1].version == vc
 
 
+    def test_download_git(self):
+        index = setuptools.package_index.PackageIndex(
+            hosts=('www.example.com',)
+        )
+
+        index._vcs_split_rev_from_url = mock.Mock()
+        url = 'https://example.com/bar'
+        rev = '2995'
+
+        index._vcs_split_rev_from_url.return_value = (url, rev)
+
+        filename = "somefile.py"
+
+        with mock.patch("os.system") as os_system_mock:
+
+            result = index._download_git(url, filename)
+
+        os_system_mock.assert_called()
+
+        assert os_system_mock.call_args_list[0][0] \
+            == ("git clone --quiet %s %s" % (url, filename), )
+
+        assert os_system_mock.call_args_list[1][0] \
+            == ("(cd %s && git checkout --quiet %s)" % (filename, rev), )
+        assert result == filename
+
+        index._vcs_split_rev_from_url.return_value = (url, None)
```

Comment:
We should make this a parametrized test, I think, but if you don't want to bother with doing that, we can make it two separate tests and it can be parametrized later.